inngest-core/src/main/kotlin/com/inngest/HttpClient.kt

```diff
@@ -0,0 +1,45 @@
+package com.inngest
+
+import com.beust.klaxon.Klaxon
+import okhttp3.Headers
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+
+typealias RequestHeaders = Map<String, String>
+
+data class RequestConfig(val headers: RequestHeaders? = null)
+
+val jsonMediaType = "application/json".toMediaType()
+
+class HttpClient(private val clientConfig: RequestConfig) {
```

Comment:
is the goal of this class to hide okhttp specific details in here as well as provide a common interface for adding headers? does it compile if we make it `internal`?

Reply:
good call, done ✔️